src/main/java/org/allenai/wikitables/WikiTablesUtil.java

```diff
@@ -0,0 +1,93 @@
+package org.allenai.wikitables;
+
+import java.util.*;
+
+import com.jayantkrish.jklol.ccg.lambda2.Expression2;
+import com.jayantkrish.jklol.ccg.lambda2.ExpressionSimplifier;
+import com.jayantkrish.jklol.ccg.lambda2.StaticAnalysis;
+import edu.stanford.nlp.sempre.Formula;
+import edu.stanford.nlp.sempre.Formulas;
+import edu.stanford.nlp.sempre.LambdaFormula;
+import fig.basic.LispTree;
+
+public class WikiTablesUtil {
+  public static String toPnpLogicalForm(Formula expression) {
+    /*
+    Sempre's lambda expressions are written differently from what pnp expects. We make the following change
+    (lambda x ((reverse fb:cell.cell.number) (var x))) -> (lambda (x) ((reverse fb:cell.cell.number) x))
+    We need to do this for all bound variables.
+     */
+    String expressionString = expression.toString();
```

Comment:
FYI I think there's a much easier way to do this if you use the utilities in Expression2:

https://github.com/jayantk/jklol/blob/master/src/com/jayantkrish/jklol/ccg/lambda2/Expression2.java

Essentially you findAll() for "lambda" then (in reverse order) use substitute to replace their argument expression with one that has parentheses. Then you do findAll for "var", then (in reverse order) use substitute on them. This method would be less brittle than the one you have here (which is sensitive to spaces and stuff).

You don't have to change this, but it's good to know for future reference.